passenger_car_dynamic_model/src/passenger_car_dynamic_model/PassengerCarDynamicModel.cpp

```diff
@@ -0,0 +1,283 @@
+    }
+  }
+
+  // Compute forces
+  if (non_zero_force_f) {
+    // Compute longitudinal slip ratio
+    const double sigma_f = isAccelerating ?
+      (no_steer_no_slip_vel - v_xc) / v_xc :   // When accelerating
+      (no_steer_no_slip_vel - v_xc) / (no_steer_no_slip_vel); // When braking
+    // Compute lateral slip angle
+    const double a_f = atan((v_yc + r * l_f_) / v_xc) + d_f;
+
+    F_xf = C_sx_ * sigma_f;
+    F_yf = -C_ay_ * a_f;
+  }
+
+  if (non_zero_force_r) {
+    // Compute longitudinal slip ratio
+    const double sigma_r = isAccelerating ?
+      (R_er_ * w_r - v_xc) / v_xc :   // When accelerating
+      (R_er_ * w_r - v_xc) / (R_er_ * w_r); // When braking
+    // Compute lateral slip angle
+    const double a_r = atan((v_yc - r * l_r_) / v_xc);
+
+    F_xr = C_sx_ * sigma_r;
+    F_yr = -C_ay_ * a_r;
+  }
+
+  const double cos_Theta = cos(Theta);
+  const double sin_Theta = sin(Theta);
+  const double cos_d_f = cos(d_f);
+  const double sin_d_f = sin(d_f);
+  
+  // Compute state_dot
+  state_dot[0] = v_xc * cos_Theta - v_yc * sin_Theta;                                   // X-dot
+  state_dot[1] = v_xc * sin_Theta + v_yc * cos_Theta;                                   // Y-dot
+  state_dot[2] = r;                                                                     // Theta-dot
+  state_dot[3] = ((F_xf * cos_d_f + F_xr - F_yf * sin_d_f) / m_) + r * v_yc;            // v_xc-dot
+  state_dot[4] = ((F_yf * cos_d_f + F_yr + F_xf * sin_d_f) / m_) - r * v_xc;            // v_yc-dot
+  state_dot[5] = (l_f_ * F_yf * cos_d_f + l_f_ * F_xf * sin_d_f - l_r_ * F_yr) / I_z_;  // r-dot
+  state_dot[6] = funcW_f(w_f, w_r, V_c);                                                // w_f-dot
+  state_dot[7] = funcW_r(w_f, w_r, V_c);                                                // w_r-dot
+  state_dot[8] = funcD_f(d_f, d_fc);                                                    // d_f-dot
+}
+
+
+void PassengerCarDynamicModel::ODEPostStep(const ODESolver::State& current, const VehicleControlInput& control, double t, const ODESolver::State& initial_state, ODESolver::State& output) const {
+  // Copy state contents
+  output = current;
+  output.resize(FULL_STATE_SIZE);
+
+  // Copy over un-simulated values
+  output[9]  = 0;
+  output[10] = control.target_steering_angle;
+  output[11] = control.target_velocity;
+}
+
+double PassengerCarDynamicModel::funcW_f(const double w_f, const double w_r, const double V_c) const {
+  return 0; // TODO Testing needs to be conducted on each vehicle to fill out this portion of the model
+}
+
+double PassengerCarDynamicModel::funcW_r(const double w_f, const double w_r, const double V_c) const {
+  return 0; // TODO Testing needs to be conducted on each vehicle to fill out this portion of the model
+}
+
+double PassengerCarDynamicModel::funcD_f(const double d_f, const double d_fc) const {
+  return 0; // TODO Testing needs to be conducted on each vehicle to fill out this portion of the model
```

Comment:
When we return 0, will our unit test case be affected?

Reply:
When we implement these functions we will need to update the unit tests. For now the unit tests have to assume constant wheel velocity.